### PR TITLE
feat: Add virtual third parties

### PR DESCRIPTION
**Pull request by LautaroPetaccio**
This PR adds the capability of using creating virtual third parties. That is, third parties created with the purpose of being temporary third parties until the registration is done on chain.

### CODE REVIEW

File: src/Item/Item.service.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { ItemCuration } from '../Curation/ItemCuration'
 import { Bridge } from '../ethereum/api/Bridge'
 import { collectionAPI } from '../ethereum/api/collection'
 import { peerAPI } from '../ethereum/api/peer'
-import { thirdPartyAPI } from '../ethereum/api/thirdParty'
+// import { thirdPartyAPI } from '../ethereum/api/thirdParty'
```
Comment (cyaiox) [inline src/Item/Item.service.ts:14]:
```suggestion

```

File: src/Item/Item.service.ts [hunk 2]
```diff
@@ -49,6 +49,7 @@ import {
   isTPItem,
   toDBItem,
 } from './utils'
+import { ThirdPartyService } from '../ThirdParty/ThirdParty.service'
```
Comment (cyaiox) [inline src/Item/Item.service.ts:52]: could you move this import above?

File: migrations/1725996635075_add-virtual-third-parties.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  // Create the trigger to update the updated_at column
+  pgm.createTrigger(
+    VirtualThirdParty.tableName,
+    update_updated_at_trigger_name,
+    {
+      when: 'BEFORE',
+      operation: 'UPDATE',
+      level: 'ROW',
+      function: update_updated_at_column_function_name,
+      condition: 'OLD.* IS DISTINCT FROM NEW.*',
```
Comment (cyaiox) [inline migrations/1725996635075_add-virtual-third-parties.ts:65]: shouldn't the updated_at field be updated with each update? or should it meet a specific condition?

File: src/ThirdParty/ThirdParty.service.ts [hunk 4]
```diff
@@ -15,18 +65,74 @@ export class ThirdPartyService {
...
+
+  static async getThirdParties(manager?: string): Promise<ThirdParty[]> {
+    const [fragments, virtualThirdParties] = await Promise.all([
+      thirdPartyAPI.fetchThirdPartiesByManager(manager),
+      manager
+        ? this.getVirtualThirdPartiesByManager(manager)
+        : ([] as ThirdParty[]),
+    ])
+    const graphThirdParties = fragments.map(toThirdParty)
+    console.log({ graphThirdParties, virtualThirdParties })
```
Comment (cyaiox) [inline src/ThirdParty/ThirdParty.service.ts:78]:
```suggestion
```

File: migrations/1725996635075_add-virtual-third-parties.ts (hunk 3 above)
Comment (LautaroPetaccio) [reply migrations/1725996635075_add-virtual-third-parties.ts:65]: The `updated_at` column will get updated only if any column in the row gets updated to something new. If it doesn't, we'll be saving a write operation by not updating the `updated_at`.

**Review by cyaiox — APPROVED** [review]

**Pull request merged**